src/manifolds/SymplecticStiefel.jl

```diff
@@ -0,0 +1,554 @@
+
+First, recall the definition the standard symplectic matrix
+````math
+Q =
+\begin{bmatrix}
+ 0    &  I \\
+-I    &  0
+\end{bmatrix}
+````
+as well as the symplectic inverse of a matrix ``A``, ``A^{+} = Q^T A^T Q``.
+
+For ``p, q ∈ \operatorname{SpSt}(2n, 2k, ℝ)`` then, we can define the
+inverse cayley retraction as long as the following matrices exist.
+````math
+    U = (I + p^+ q)^{-1} \in ℝ^{2k \times 2k},
+    \quad
+    V = (I + q^+ p)^{-1} \in ℝ^{2k \times 2k}.
+````
+
+If that is the case, the inverse cayley retration at ``p`` applied to ``q`` is
+````math
+\mathcal{L}_p^{\operatorname{Sp}}(q) = 2p\bigl(V - U\bigr) + 2\bigl((p + q)U - p\bigr)
+                                        ∈ T_p\operatorname{Sp}(2n).
+````
+
+[^Bendokat2021]:
+    > Bendokat, Thomas and Zimmermann, Ralf:
+	> The real symplectic Stiefel and Grassmann manifolds: metrics, geodesics and applications
+	> arXiv preprint arXiv:2108.12447, 2021 (https://arxiv.org/abs/2108.12447)
+"""
+function inverse_retract!(M::SymplecticStiefel, X, p, q, ::CayleyInverseRetraction)
+    U_inv = lu!(add_scaled_I!(symplectic_inverse_times(M, p, q), 1))
+    V_inv = lu!(add_scaled_I!(symplectic_inverse_times(M, q, p), 1))
+
+    X .= 2 .* ((p / V_inv .- p / U_inv) .+ ((p + q) / U_inv) .- p)
+    return X
+end
+
+function gradient(M::SymplecticStiefel, f, p, backend::RiemannianProjectionBackend)
+    amb_grad = _gradient(f, p, backend.diff_backend)
+    return grad_euclidean_to_manifold(M, p, amb_grad)
+end
+
+function gradient!(M::SymplecticStiefel, f, X, p, backend::RiemannianProjectionBackend)
+    _gradient!(f, X, p, backend.diff_backend)
+    return grad_euclidean_to_manifold!(M, X, p, copy(X))
+end
+
+function grad_euclidean_to_manifold(M::SymplecticStiefel, p, ∇f_euc)
+    return grad_euclidean_to_manifold!(M, similar(∇f_euc), p, ∇f_euc)
+end
+
+function grad_euclidean_to_manifold!(::SymplecticStiefel, ∇f_man, p, ∇f_euc)
+    Q = SymplecticMatrix(p, ∇f_euc)
+    Qp = Q * p                      # Allocated memory.
+    mul!(∇f_man, ∇f_euc, (p' * p))  # Use the memory in ∇f_man.
+    ∇f_man .+= Qp * (∇f_euc' * Qp)
+    return ∇f_man
+end
+
+@doc raw"""
+    project!(::Union{SymplecticStiefel, Symplectic}, Y, p, A)
+
+Given a point ``p \in \operatorname{SpSt}(2n, 2k)``,
+project an element ``A \in \mathbb{R}^{2n \times 2k}`` onto
+the tangent space ``T_p\operatorname{SpSt}(2n, 2k)`` relative to
+the euclidean metric of the embedding ``\mathbb{R}^{2n \times 2k}``.
+
+That is, we find the element ``X \in T_p\operatorname{SpSt}(2n, 2k)``
+which solves the constrained optimization problem
+
+````math
+    \operatorname{min}_{X \in \mathbb{R}^{2n \times 2k}} \frac{1}{2}||X - A||^2, \quad
+    \text{s.t.}\;
+    h(X) = X^T Q p + p^T Q X = 0,
+````
+where ``h : \mathbb{R}^{2n \times 2k} \rightarrow \operatorname{skew}(2k)`` defines
+the restriction of ``X`` onto the tangent space ``T_p\operatorname{SpSt}(2n, 2k)``.
+"""
+function project!(::Union{SymplecticStiefel,Symplectic}, Y, p, A)
```

Comment:
We usually document the `project` function.

Reply:
Done.